pkg/planner/core/find_best_task.go

```diff
@@ -540,6 +540,14 @@ func findBestTask(lp base.LogicalPlan, prop *property.PhysicalProperty, planCoun
 		planCounter.Dec(1)
 		return bestTask, 1, nil
 	}
+	// if prop is require an index join's probe side, check the inner pattern admission here.
```

Comment:
baseLogicalPlan's findBestTask portal should control the index join prop admission.